tests/core/engine_adapter/integration/test_integration.py

```diff
@@ -1328,31 +1306,39 @@ def test_sushi(ctx: TestContext):
         personal_paths=[pathlib.Path("~/.sqlmesh/config.yaml").expanduser()],
     )
 
+    current_gateway_config = config.gateways[ctx.gateway]
+    current_gateway_config.state_schema = sushi_state_schema
+    if (sc := current_gateway_config.state_connection) and sc.type_ == "duckdb":
+        current_gateway_config.connection.concurrent_tasks = 1  # duckdb cant handle parallelism
+
     if ctx.dialect == "athena":
         # Ensure that this test is using the same s3_warehouse_location as TestContext (which includes the testrun_id)
-        config.gateways[
-            ctx.gateway
-        ].connection.s3_warehouse_location = ctx.engine_adapter.s3_warehouse_location
-
-    # clear cache from prior runs
-    cache_dir = pathlib.Path("./examples/sushi/.cache")
-    if cache_dir.exists():
-        import shutil
-
-        shutil.rmtree(cache_dir)
+        current_gateway_config.connection.s3_warehouse_location = (
+            ctx.engine_adapter.s3_warehouse_location
+        )
 
-    context = Context(paths="./examples/sushi", config=config, gateway=ctx.gateway)
+    # Copy sushi example to tmpdir
+    shutil.copytree(pathlib.Path("./examples/sushi"), tmp_path, dirs_exist_ok=True)
+
+    # Rewrite schema references to test schema references
+    # Note that we deliberately do it at the filesystem level instead of messing with the Context to ensure
+    # that we are testing an actual Context rather than a doctored one
+    extensions = ["*.sql", "*.yaml", "*.py"]
+    replacements = {
```

Comment:
I'm open to a better way of making the `sushi` schema dynamic instead of using text-based find/replace.

I didn't want to modify the example to use vars in all the model names just for these tests because I think there is value in keeping our examples reasonably simple

Reply:
Yeah this is annoying because of our decision to use the examples as our test fixtures. To be fair we have already made decisions in the past that adds complexity to sushi just for test sake but I get how you could feel this would cross a line.

Reply:
Well if we don't mind adding more complexity to Sushi I could use it to demonstrate (and test I guess) dynamic model names.

For example, configuring a global variable `env: prod`, changing the model names to eg `sushi_@{env}.customer_total_revenue` and then in tests setting `env` to the test_id.

Do you think that's a better solution?

Reply:
I agree with your original assessment that readability/simplicity of the example is better than making this more robust. Therefore I am good with what you have. I was pointing out though that in other cases we have decided to add complexity to sushi so we are inconsistent.